### PR TITLE
Market: Tests for CronTick

### LINKED ISSUE #574: Market: Tests for VerifiedDeals in PublishStorageDeals
In the market actor, we need to understand and test the code that talks to the `VerifiedRegistryActorAddr` in `PublishStorageDeals`.

@anorth FYI. Please give the labels.

### LINKED ISSUE #613: Market: Test Send to VerifiedRegistryActorAddr for timed-out verified deals in CronTick
https://github.com/filecoin-project/specs-actors/blob/a0ab84303c0ec6694001929e5442f1d7b9c2491e/actors/builtin/market/market_actor.go#L615 needs to be tested.

### PR DESCRIPTION
**Pull request by aarshkshah1992**
Tests for the CronCallback.

Also Closes #613 .
Also Closes #574 .

### CODE REVIEW

File: actors/builtin/market/market_test.go [hunk 1]
```diff
@@ -1093,6 +1094,579 @@ func TestOnMinerSectorsTerminate(t *testing.T) {
...
+
+func TestCronTickDealSlashing(t *testing.T) {
+	owner := tutil.NewIDAddr(t, 101)
+	provider := tutil.NewIDAddr(t, 102)
+	worker := tutil.NewIDAddr(t, 103)
+	client := tutil.NewIDAddr(t, 104)
+	mAddrs := &minerAddrs{owner, worker, provider}
+	sectorExpiry := abi.ChainEpoch(400)
+
+	// hairy edge cases
```
Comment (acruikshank) [inline actors/builtin/market/market_test.go:1438]: This is fantastic!

File: actors/builtin/market/market_test.go [hunk 2]
```diff
@@ -1093,6 +1094,579 @@ func TestOnMinerSectorsTerminate(t *testing.T) {
...
+					})
+				}
+			})
+		}
+	}
+
+	startEpoch := abi.ChainEpoch(50)
+	endEpoch := abi.ChainEpoch(300)
+
+	{
```
Comment (acruikshank) [inline actors/builtin/market/market_test.go:1549]: Not clear why this is here. The extra indentation hurts readability a bit.

File: actors/builtin/market/market_test.go [hunk 3]
```diff
@@ -1093,6 +1094,579 @@ func TestOnMinerSectorsTerminate(t *testing.T) {
...
+
+	t.Run("crontick for a deal at it's start epoch results in zero payment and no slashing", func(t *testing.T) {
+		rt, actor := basicMarketSetup(t, owner, provider, worker, client)
+		dealId := actor.publishAndActivateDeal(rt, client, mAddrs, startEpoch, endEpoch, 0, sectorExpiry)
+
+		// move the current epoch to startEpoch
+		current := startEpoch
+		rt.SetEpoch(current)
+		pay, slashed := actor.cronTickAndAssertBalances(rt, client, provider, current, dealId)
+		require.EqualValues(t, pay, big.Zero())
```
[the remark this answers is not in the document]
Comment (aarshkshah1992) [reply actors/builtin/market/market_test.go]: This is done.

File: actors/builtin/market/market_test.go [hunk 4]
```diff
@@ -1093,6 +1094,579 @@ func TestOnMinerSectorsTerminate(t *testing.T) {
...
+		require.EqualValues(t, big.Zero(), slashed)
+
+		actor.getDealProposal(rt, dealId)
```
[the remark this answers is not in the document]
Comment (aarshkshah1992) [reply actors/builtin/market/market_test.go]: Basically wanted to assert that the deals aren't deleted. Have fixed this now.

File: actors/builtin/market/market_test.go (hunk 2 above)
Comment (aarshkshah1992) [reply actors/builtin/market/market_test.go]: Fixed this.

File: actors/builtin/market/market_actor.go [hunk 5]
```diff
@@ -526,14 +514,12 @@ func (a Actor) CronTick(rt Runtime, params *adt.EmptyValue) *adt.EmptyValue {
 				if err != nil {
 					return xerrors.Errorf("failed to get cid for deal proposal: %w", err)
 				}
-				if err := pending.Delete(adt.CidKey(dcid)); err != nil {
-					builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to delete pending proposal")
-				}
+
+				builtin.RequireNoErr(rt, pending.Delete(adt.CidKey(dcid)), exitcode.ErrIllegalState, "failed to delete pending proposal")
```
Comment (whyrusleeping) [inline actors/builtin/market/market_actor.go:518]: @anorth I think writing things inline like this is fairly difficult to read, we should probably at least break things like this into two lines.

Comment (anorth) [reply actors/builtin/market/market_actor.go:518]: Yes, I agree. The `pending.Delete` call would be better on its own line.

Comment (aarshkshah1992) [reply actors/builtin/market/market_actor.go:518]: @anorth This has been fixed.

**Pull request merged**